benchmark_data_loading/dataset_helpers.py

```diff
@@ -102,28 +103,51 @@ def _make_dp_from_tars(*, root, archive_size):
     return dp
 
 
-def _make_webdataset(*, root, archive_size):
-    archives = Path(root).glob(f"archive_{archive_size}*.tar")
-    archives = [str(a) for a in archives]
-    return wds.WebDataset(archives)  # This will read and load the data as bytes
-
-
 def make_dp(*, root, archive=None, archive_content=None, archive_size=500):
     if archive in ("pickle", "torch"):
         dp = _make_dp_from_archive(
             root=root, archive=archive, archive_content=archive_content, archive_size=archive_size
         )
     elif archive == "tar":
         dp = _make_dp_from_tars(root=root, archive_size=archive_size)
-    elif archive == "webdataset":
-        dp = _make_webdataset(root=root, archive_size=archive_size)
     else:
         dp = _make_dp_from_image_folder(root=root)
 
     dp = LenSetter(dp)
     return dp
 
 
+def make_webdataset_dataloader(*, root, encoded=True, archive_size=500):
+    def preprocess(sample):
+        encoded_image = bytesio_to_tensor(io.BytesIO(sample["jpeg"]))
+        label = int(str(sample["__key__"].split(os.sep)[0]))
+        return encoded_image, label
+
+    archives = Path(root).glob(f"archive_{archive_size}*.tar")
+    archives = [str(a) for a in archives]
+
+    dataset = wds.WebDataset(archives).map(preprocess)
```

Comment:
Instead of pre-processing here I think it'll be best to do that outside of this function, which should probably just return a simple WebDataset with the raw archive content - like what the other dataset builders are doing.

This way we can get fair comparisons in `bench_data_reading.py` (potentially using `bytesio_to_tensor` there). And in then in ` `bench_data_reading_decoding.py` we can apply the decoding step, and finally in `with_DL` we can wrap it in a `WebLoader`. Does that make sense?